lib/primitive/lib/Cardano/Wallet/Primitive/Types/Certificates.hs

```diff
@@ -76,6 +80,8 @@ data DelegationCertificate
     = CertDelegateNone RewardAccount
     | CertDelegateFull RewardAccount PoolId
     | CertRegisterKey RewardAccount
+    | CertVoteFull RewardAccount VoteAction
+    | CertDelegateAndVoteFull RewardAccount PoolId VoteAction
```

Comment:
```suggestion
    | CertVoteAndDelegate RewardAccount (Maybe DRep) (Maybe PoolId)
```

The formal specification for Conway uses the following constructor for the `DCert` state transition datum:

<img width="613" alt="DCert" src="https://github.com/cardano-foundation/cardano-wallet/assets/564592/a2e63303-eadb-4b68-a488-66e98b8918ef">

Here, both voting and delegation are joined into a single constructor.

Again, in order to stick to existing naming conventions, I would follow their example here.

Using a single constructor is likely to slightly simplify our code. We could remove the `CertDelegateFull` constructor as well, but that would involve changes to our code that are beyond the scope of this pull request.

Reply:
used CertVoteAndDelegate and removed other two that are handled here after adding optional pool and vote